sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -621,6 +620,8 @@ object DataSource extends Logging {
         "org.apache.spark.sql.hive.orc.OrcFileFormat"
       case "com.databricks.spark.avro" if conf.replaceDatabricksSparkAvroEnabled =>
         "org.apache.spark.sql.avro.AvroFileFormat"
+      case name if name.equalsIgnoreCase("none") =>
+        "org.apache.spark.sql.execution.datasources.none.NoneDataSource"
```

Comment:
Ur, do we need to add here? I expected `sql/core/src/main/resources/META-INF/services/org.apache.spark.sql.sources.DataSourceRegister`.

Reply:
This place is only used to handle exceptional data source aliases like `native` or `avro`.

Reply:
+1. other test sources like "rate" is also there.

Reply:
Yea, I do think so.

Reply:
+1

Reply:
done